tests/src/Cas/Protocol/SamlValidateTest.php

```diff
@@ -27,24 +26,43 @@ public function testSamlValidatXmlGeneration(): void
 
         $samlValidate = new SamlValidateResponder();
         $xmlString = $samlValidate->convertToSaml($ticket);
-        $response = new SimpleXMLElement(strval($xmlString));
```

Comment:
what is the goal with switching from SimpleXMLElement?

Reply:
SimpleXMLElement was not parsing the xml document correctly since part of the document has namespaces and part does not. I used the `xpath` method to get the nodes i needed to test but still i was getting errors. Switching to `xml_parse_into_struct` made things really easy since the method does not care about namespaces. It cares about node names and attributes. Then pushes everything into two arrays.